src/Features/Core/Portable/CodeFixes/FixAllOccurrences/IFixMultipleOccurrencesService.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.CodeFixes
+{
+    internal interface IFixMultipleOccurrencesService : IWorkspaceService
+    {
+        /// <summary>
+        /// Computes the fix muliple occurrences code fix for the given diagnostics with source locations, brings up the preview changes dialog for the fix and
+        /// apply the code action operations corresponding to the fix.
+        /// </summary>
+        void ComputeAndApplyFix(
+            ImmutableDictionary<Document, ImmutableArray<Diagnostic>> diagnosticsToFix,
+            Workspace workspace,
+            CodeFixProvider fixProvider,
+            FixAllProvider fixAllProvider,
+            string equivalenceKey,
+            string previewChangesTitle,
+            string waitDialogMessage,
+            CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Computes the fix muliple occurrences code fix for the given diagnostics without any source location, brings up the preview changes dialog for the fix and
+        /// apply the code action operations corresponding to the fix.
+        /// </summary>
+        void ComputeAndApplyFix(
+            ImmutableDictionary<Project, ImmutableArray<Diagnostic>> diagnosticsToFix,
+            Workspace workspace,
+            CodeFixProvider fixProvider,
+            FixAllProvider fixAllProvider,
+            string equivalenceKey,
+            string previewChangesTitle,
+            string waitDialogMessage,
+            CancellationToken cancellationToken);
+    }
+}
```

Comment:
You are right in both observations. This only adds support for fixing multiple diagnostics for a single fixer, probably can be extended for multi-fixer case.

Our SuggestedAction implementation now works fine in absence of a source subject buffer, so probably it can work fine as soon as LB tries to invoke such an action. Probably @shyamnamboodiripad knows better if that will suffice when there is LB support in error list.